### PR TITLE
Added onError callback if external command fails.

### PR DESCRIPTION
**Pull request by badyto**
This is needed for example to be able to log external commands that failed (unhandled exceptions in external programs for example).

### CODE REVIEW

File: src/GO/Job.php [hunk 1]
```diff
@@ -361,7 +375,8 @@ public function run()
         if (is_callable($compiled)) {
             $this->output = $this->exec($compiled);
         } else {
-            $this->output = exec($compiled);
+            $outputArray = null;
+            $this->output = exec($compiled, $outputArray, $this->returnCode);
```
Comment (peppeocchi) [inline src/GO/Job.php]: Will this force every job to run in foreground?
It seems like a few tests are now failing, the tests that ensure that jobs can run in background.

**Comment by peppeocchi** [comment]
@badyto thank you, it's a reasonable request, can you please check the comment I left?
Did you had a look at the [then method](https://travis-ci.org/peppeocchi/php-cron-scheduler/builds/261005010?utm_source=github_status&utm_medium=notification) to achieve the same result?

It seems like a few tests are now failing, if I remember correctly when passing the other two arguments to the `exec` function you're basically ignoring the functionality to run jobs in background.

**Comment by badyto** [comment]
It is very strange, the tests fail only for PHP7.0 and only here on travis-ci. I have tried to run the tests on my machine both on PHP7.0 and PHP7.1 and they all pass... Furthermore - the one job that failed here on travis-ci says 192 assertions, while the other two tests say 194 assertions. My local machine says 194 assertions for all PHP versions. Is it possible there is something fishy going on with the travis-ci...?

**Comment by peppeocchi** [comment]
Yes it's a bit weird, I've done a few tests myself and it seems the jobs will keep running in background when passing the other two params to `exec`.

I've tested a different implementation of your PR, using the [then method]() to handle the script error.
I created that method on v2 because I wanted to leverage functionalities like this, keeping the scheduler as light as possible (because of that, I had to remove a great log integration (https://github.com/peppeocchi/php-cron-scheduler/pull/4) written by @carstenwindler)

To give you an example, this is how it should look like
```php
$onError = function () {
    // handle the script error
};

$scheduler->php('script.php')->then(function ($output, $returnCode) use ($onError) {

    if ($returnCode) {
        $onError();
    }

});
```

I've reused your test to prove that the end result is the same.

Please have a look at this commit https://github.com/peppeocchi/php-cron-scheduler/commit/77e2e95ab114d71521dd69555c7d496b44e01a53, if you want to add any other change please send it to branch [v2.x](https://github.com/peppeocchi/php-cron-scheduler/tree/v2.x)

**Pull request closed without merging**